src/test_c_kzg_4844.c

```diff
@@ -486,6 +743,103 @@ static void test_reverse_bits__succeeds_all_bits_are_one(void) {
     ASSERT_EQUALS(reverse_bits(original), reversed);
 }
 
+///////////////////////////////////////////////////////////////////////////////
+// Tests for bit_reversal_permutation
+///////////////////////////////////////////////////////////////////////////////
+
+static void test_bit_reversal_permutation__succeeds_round_trip(void) {
+    C_KZG_RET ret;
+    uint32_t original[128];
+    uint32_t reversed_reversed[128];
+
+    for (size_t i = 0; i < 128; i++) {
+        get_rand_uint32(&original[i]);
+        reversed_reversed[i] = original[i];
+    }
+    ret = bit_reversal_permutation(&reversed_reversed, sizeof(uint32_t), 128);
+    ASSERT_EQUALS(ret, C_KZG_OK);
+    ret = bit_reversal_permutation(&reversed_reversed, sizeof(uint32_t), 128);
+    ASSERT_EQUALS(ret, C_KZG_OK);
+    for (size_t i = 0; i < 128; i++) {
+        ASSERT_EQUALS(reversed_reversed[i], original[i]);
+    }
+}
+
+static void test_bit_reversal_permutation__specific_items(void) {
+    C_KZG_RET ret;
+    uint32_t original[128];
+    uint32_t reversed[128];
+
+    for (size_t i = 0; i < 128; i++) {
+        get_rand_uint32(&original[i]);
+        reversed[i] = original[i];
+    }
+    ret = bit_reversal_permutation(&reversed, sizeof(uint32_t), 128);
+    ASSERT_EQUALS(ret, C_KZG_OK);
+
+    ASSERT_EQUALS(reversed[0], original[0]);
```

Comment:
Could you add a short comment here reminding the reader that it's reordering based on array indices and not the array values? At first, I thought this was a position-dependent test because I forgot how it worked.